dali/kernels/imgproc/resample/resampling_batch.cu

```diff
@@ -0,0 +1,157 @@
+    in_shape = sample.tmp_shape;
+    out_shape = sample.out_shape;
+  }
+  int block = sbi.block_in_sample;
+
+  // 0 = vertical axis, 1 = horizontal axis
+  int axis = (which_pass == 0) ^ (sample.order == SeparableResamplingSetup::VertHorz);
+
+  ResamplingFilterType ftype = sample.filter_type[axis];
+  ResamplingFilter filter = sample.filter[axis];
+  int support = filter.support();
+
+  float scale = in_shape[axis] * (1.0f / out_shape[axis]);
+
+  int block_size = axis == 1 ? blockDim.x : blockDim.y;
+  int size_in_blocks = (out_shape[axis] + block_size - 1) / block_size;
+
+  int start = min(size_in_blocks * block / blocks * block_size, out_shape[axis]);
+  int end   = min(size_in_blocks * (block + 1) / blocks * block_size, out_shape[axis]);
+
+  int x0, x1, y0, y1;
+
+  if (axis == 1) {
+    x0 = start;
+    x1 = end;
+    y0 = 0;
+    y1 = out_shape[0];
+  } else {
+    x0 = 0;
+    x1 = out_shape[1];
+    y0 = start;
+    y1 = end;
+  }
+
+  switch (ftype) {
+  case ResamplingFilterType::Nearest:
+    if (axis == 1) {
+      NNResample(x0, x1, y0, y1, 0, 0, scale, 1, sample_out, out_stride, sample_in, in_stride,
+        in_shape[1], in_shape[0], sample.channels);
+    } else {
+      NNResample(x0, x1, y0, y1, 0, 0, 1, scale, sample_out, out_stride, sample_in, in_stride,
+        in_shape[1], in_shape[0], sample.channels);
+    }
+    break;
+  case ResamplingFilterType::Linear:
+    if (axis == 1) {
+      LinearHorz(x0, x1, y0, y1, 0, scale, sample_out, out_stride, sample_in, in_stride,
+        in_shape[1], sample.channels);
+    } else {
+      LinearVert(x0, x1, y0, y1, 0, scale, sample_out, out_stride, sample_in, in_stride,
+        in_shape[0], sample.channels);
+    }
+    break;
+  default:
+    if (axis == 1) {
+      ResampleHorz(x0, x1, y0, y1, 0, scale, sample_out, out_stride, sample_in, in_stride,
+        in_shape[1], sample.channels, filter, support);
+    } else {
+      ResampleVert(x0, x1, y0, y1, 0, scale, sample_out, out_stride, sample_in, in_stride,
+        in_shape[0], sample.channels, filter, support);
+    }
+    break;
+  }
+}
+
+template <int which_pass, typename Output, typename Input>
+void BatchedSeparableResample(Output *out, const Input *in,
+  const SeparableResamplingSetup::SampleDesc *samples,
+  int num_samples, const SampleBlockInfo *block2sample, int num_blocks,
+  cudaStream_t stream) {
+
+  if (num_blocks <= 0)
+    return;
+
+  dim3 blockSize(32, 24);
+
+  BatchedSeparableResampleKernel<which_pass>
+  <<<num_blocks, blockSize, ResampleSharedMemSize, stream>>>(
+    out, in, samples, block2sample);
+}
+
+
+#define INSTANTIATE_BATCHED_RESAMPLE(which_pass, Output, Input) \
+template void BatchedSeparableResample<which_pass, Output, Input>( \
+  Output *out, const Input *in, \
+  const SeparableResamplingSetup::SampleDesc *samples, \
+  int num_samples, const SampleBlockInfo *block2sample, int num_blocks, \
+  cudaStream_t stream)
+
+INSTANTIATE_BATCHED_RESAMPLE(0, float, uint8_t);
```

Comment:
TODO: uncomment for final build - it simply took too long to rebuild when testing...